tests/containers/secret.pm

```diff
@@ -0,0 +1,89 @@
+
+    # Create a secret2 from CLI and inspect it
+    record_info("secret create CLI");
+    assert_script_run("printf T0p_S3cr3t2 | $runtime secret create secret2 -", fail_message => "Error creating secret from CLI", timeout => 60);
+    record_info("secret inspect CLI", script_output("$runtime secret inspect secret2"));
+
+    # Check if secret exists (only in podman)
+    if ($runtime =~ 'podman') {
+        record_info("secret exists");
+        assert_script_run("podman secret exists secret1", fail_message => "Error checking if secret exists");
+        assert_script_run("podman secret exists secret2", fail_message => "Error checking if secret exists");
+        # This secret3 does not exist and thus `secret exists` must return 1
+        assert_script_run("! podman secret exists secret3", fail_message => "Error checking if secret exists");
+    }
+
+    # List all secrets
+    record_info("secret ls");
+    assert_script_run("$runtime secret ls");
+
+    # Run two containers passing secret1 as an env variable and secret2 as default
+    my $runtime_command = ($runtime =~ 'docker') ? 'service create' : 'run';
+    record_info("Access secrets");
+    script_run("$runtime pull registry.opensuse.org/opensuse/bci/bci-busybox:latest", timeout => 120);
+
+    # secret1 testing (default secret)
+    validate_script_output("$runtime $runtime_command --name secret1-test --secret secret1 bci-busybox:latest cat /run/secrets/secret1", sub { m/T0p_S3cr3t1/ });
+    # Commit the container and check that the secrets are not in it
+    record_info("Commit container secret1-test");
+    assert_script_run("$runtime commit secret1-test secret1-test-image");
+    my $output = script_output("$runtime $runtime_command --name secret1-test-commit secret1-test-image:latest cat /run/secrets/secret1", proceed_on_failure => 1);
+    die("Secret commited") if ($output =~ !m/T0p_S3cr3t1/);
+
+    # Accessing secrets as env variables is not available in Docker
+    if ($runtime =~ 'podman') {
+        # secret2 testing (env secret)
+        validate_script_output("podman run --name secret2-test --secret secret2,type=env,target=TOP_SECRET2 bci-busybox:latest printenv TOP_SECRET2", sub { m/T0p_S3cr3t2/ });
+        # Commit the container and check that the secrets are not in it
+        record_info("Commit container secret2-test");
+        assert_script_run("podman commit secret2-test secret2-test-image");
+        $output = script_output("podman run --name secret2-test-commit secret2-test-image:latest printenv TOP_SECRET2", proceed_on_failure => 1);
+        die("Secret commited") if ($output =~ !m/T0p_S3cr3t2/);
+    }
+
+    # Remove secrets
+    record_info("secret rm");
+    assert_script_run("$runtime secret rm secret1 secret2");
+    validate_script_output("$runtime secret ls --quiet", sub { m// }, fail_message => "Secrets have not been deleted");
+
+    # Stop the swarm in Docker
+    assert_script_run("docker swarm leave --force") if ($runtime == 'docker');
```

Comment:
Sometimes you use `$runtime == 'docker'` and some other times `$runtime =~ 'docker'`. I would use `==` in all cases, as we don't expect other strings than `docker|podman`.